### PR TITLE
ROC-2892: Agreement countersigning

### PR DESCRIPTION
**Pull request by radoslawn-kainos**
### JIRA link

[2892](https://tools.hmcts.net/jira/browse/ROC-2892)

### Change description

Add agreement countersigning

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No
- [x] No idea

### CODE REVIEW

File: src/main/features/offer/form/models/declaration.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  constructor (signed?: boolean) {
+    this.signed = signed
+  }
+
+  static fromObject (value?: any): Declaration {
+    if (!value) {
+      return value
+    }
+    return new Declaration(value.signed === 'true')
```
Comment (timja-kainos) [inline src/main/features/offer/form/models/declaration.ts]: could we use `toBoolean` here please?
`import * as toBoolean from 'to-boolean'`

File: src/main/features/offer/views/countersign-agreement.njk [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      </p>
+
+      <p>
+        {{ t('The agreement is a legal contract between you and {{name}}.', {
+               name: claim.claimData.defendant.name
+             })
+        }}     
+      <p>
+      
+      <a href="{{ OfferPaths.declarationPage.evaluateUri({ externalId: claim.externalId }) }}" class="button">
```
Comment (timja-kainos) [inline src/main/features/offer/views/countersign-agreement.njk]: I think if we're pretending things are buttons then we're supposed to add a `role="button"` to the markup there's a shim that then makes them click when you press enter on them

File: src/main/features/offer/views/declaration.njk [hunk 3]
```diff
@@ -12,31 +11,82 @@
...
+      <p>
+        {{
+          t('<a href="{{ n244formUri }}">Complete form N244</a> and send it to the <a href="{{ countyCourtBusinessCentreUri }}">County Court Business Centre</a>. You need to say on the form which of the 2 options you’re applying for.', {
+            n244formUri: '#TODO',
+            countyCourtBusinessCentreUri: '#TODO'
+          }) | safe
+        }}
+      </p>
+
+      <p>{{ t('The original claim and any counterclaims will be kept on hold in case this agreement is broken. They will be closed on [date] if we don’t hear from either of you.') }}</p>
```
Comment (timja-kainos) [inline src/main/features/offer/views/declaration.njk]: don't forget to update the date

File: src/main/features/offer/views/settled.njk [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{% extends "layout.njk" %}
+{% from "support.njk" import supportWidget %}
+
+{% set title = t('Money claim | Claim settled - confirmation page') %}
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <div class="govuk-box-highlight">
+        <h1 class="bold-large">{{ t('You’ve settled out of court') }}</h1>
```
Comment (timja-kainos) [inline src/main/features/offer/views/settled.njk]: shouldn't this be heading-large not bold-large class?
REQ

Comment (radoslawn-kainos) [reply src/main/features/offer/views/settled.njk]: Done

File: src/main/features/offer/views/countersign-agreement.njk (hunk 2 above)
Comment (radoslawn-kainos) [reply src/main/features/offer/views/countersign-agreement.njk]: Makes sense, done

File: src/main/app/claims/offerClient.ts [hunk 5]
```diff
@@ -37,4 +37,13 @@ export class OfferClient {
       }
     })
   }
+
+  static countersignOffer (externalId: string, user: User): Promise<Claim> {
+    return request.post(`${claimStoreApiUrl}/${externalId}/offers/defendant/countersign`, {
+      body: '',
```
Comment (timja-kainos) [inline src/main/app/claims/offerClient.ts]: do you need to specify body here?

File: src/main/features/offer/views/accepted.njk [hunk 6]
```diff
@@ -20,14 +16,17 @@
...
+            dashboardUri: '/dashboard'
+          }) | safe
+        }}
       </p>
 
       {{ supportWidget() }}
 
       <br>
-      <a href="#" class="button">Finished</a>
+      <a href="#" role="button" class="button">Finished</a>
```
Comment (timja-kainos) [inline src/main/features/offer/views/accepted.njk]: shouldn't this link to an end of service survey?

File: src/main/features/offer/views/countersign-agreement.njk [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+
+      <h1 class="heading-large">{{ t('Make a legal agreement') }}</h1>
+
+      <p>
+        {{ t('You can make a legal agreement to settle out of court.') }}
+        {{ t('It’s based on the information you and {{name}} have provided. It has not been reviewed by the court.', {
```
Comment (timja-kainos) [inline src/main/features/offer/views/countersign-agreement.njk]: spaces around `{{name}}` please applies to other places in this file too

File: src/main/features/offer/views/settled.njk [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+          t('View your claim status on <a href="{{ dashboardUri }}">your money claims account</a>.', {
+            dashboardUri: '/dashboard'
+          }) | safe
+        }}
+      </p>
+
+      {{ supportWidget() }}
+
+      <br>
+      <a href="#" role="button" class="button">Finished</a>
```
Comment (timja-kainos) [inline src/main/features/offer/views/settled.njk]: shouldn't this link to an end of service survey?

File: src/main/app/claims/offerClient.ts (hunk 5 above)
Comment (radoslawn-kainos) [reply src/main/app/claims/offerClient.ts]: No idea, but other methods did, so I did the same here.

**Pull request merged**